reV/rep_profiles/rep_profiles.py

```diff
@@ -343,6 +340,50 @@ def __init__(self, gen_fpath, rev_summary, gid_col='gen_gids',
         self._err_method = err_method
         self._weight = weight
         self._n_profiles = n_profiles
+        self._gen_gids = None
+        self._res_gids = None
+
+        self._init_profiles_weights()
+
+    def _init_profiles_weights(self):
+        """Initialize the base source profiles and weight arrays"""
+        gen_gids = self._get_region_attr(self._rev_summary, 'gen_gids')
+        res_gids = self._get_region_attr(self._rev_summary, 'res_gids')
+
+        self._weights = np.ones(len(res_gids))
+        if self._weight is not None:
+            self._weights = self._get_region_attr(self._rev_summary,
+                                                  self._weight)
+
+        df = pd.DataFrame({'gen_gids': gen_gids,
+                           'res_gids': res_gids,
+                           'weights': self._weights})
+        df = df.sort_values('res_gids')
+        self._gen_gids = df['gen_gids'].values
+        self._res_gids = df['res_gids'].values
+        if self._weight is not None:
+            self._weights = df['weights'].values
+        else:
+            self._weights = None
+
+        with Resource(self._gen_fpath) as res:
+            meta = res.meta
+
+            assert 'gid' in meta
+            source_res_gids = meta['gid'].values
+            msg = ('Resource gids from "gid" column in meta data from "{}" '
+                   'must be sorted! reV generation should always be run with '
+                   'sequential project points.'.format(self._gen_fpath))
+            assert np.all(source_res_gids[:-1] <= source_res_gids[1:]), msg
```

Comment:
Is there any harm in just sorting them in on the fly?

Reply:
I think it is actually a requirement of reV gen, with the only exception being if you use gid_map and to not replace the true gid's with the mapped ones. I'm just doing an assert here because the data read logic requires this so it's worth double checking (could prevent a really annoying bug down the road). You can't sort the meta because then the extraction from the h5 would be out of order... you could read the whole h5 and then sort the columns but that would be onerous.

Reply:
Ah gotcha, I think you are right @grantbuster. Looks like the project points input to reV gen is sorted if it is out of order, so it is basically impossible to create out-of-order "gids" unless you intentionally manipulate the data outside of the reV pipeline (in which case that's on you 😛). Thanks for clearing this up for me

Reply:
Exactly, so the check is more of a "all of the subsequent logic depends on this, make sure we have a in-situ test just in case this assumption changes for some reason"